api/controller.py

```diff
@@ -1075,6 +1076,34 @@ def search(self, lane_identifier, feed_class=AcquisitionFeed):
             facets=facets
         )
 
+    def qa_feed(self, feed_class=AcquisitionFeed):
+        """Create an OPDS feed containing the information necessary to
+        run a full set of integration tests against this server and
+        the vendors it relies on.
+        """
+
+        library = flask.request.library
+        search_engine = self.search_engine
+        if isinstance(search_engine, ProblemDetail):
+            return search_engine
+
+        url = self.url_for(
+            "qa_feed",
+            library_short_name=library.short_name,
+        )
+
+        jwl = JackpotWorkList(library)
+        annotator = self.manager.annotator(jwl)
+
+        # TODO: Make it possible to pass a pagination object into
+        # groups() to override the standard size of a grouped lane,
+        # improving performance.
+        return feed_class.groups(
+            _db=self._db, title="QA test feed", url=url, worklist=jwl,
+            annotator=annotator, search_engine=search_engine,
+            max_age=CachedFeed.IGNORE_CACHE
```

Comment:
Would it make sense to cache the generated QA feed by default and just add an additional boolean parameter `reset_cache` to `/feed/qa`?

Reply:
Good question. I talked to @vasilyevi about this and he believes it will only be necessary to fetch this feed once per test run, not once for each individual test. If that's true, then it should be OK to avoid special caching code.

However thinking about this, it's so expensive to generate this feed, that I would like to cache it for a few minutes just to speed up the process of developing the tests. Adding caching here creates an interaction with the `cachedfeeds` table, so it will require a little more work. I'll look into this today and maybe I'll be able to speed up the feed generation as well.

I don't want to cache for too long because the whole point of this feed is all these books are available right now. Running the test suite could put some of these books in the "unavailable" state and invalidate the feed.